crates/iota-core/src/transaction_input_loader.rs

```diff
@@ -134,10 +136,11 @@ impl TransactionInputLoader {
         &self,
         shared_lock_store: &impl GetSharedLocks,
         tx_key: &TransactionKey,
+        _tx_lock: &CertLockGuard, // see below for why this is needed
```

Comment:
I don't see anything below?

Reply:
The reason is the following, as included in this PR, and same as the MystenLabs/sui@3aac32e
```rust
                            // Important to hold the _tx_lock here - otherwise it would be possible
                            // for a concurrent execution of the same tx to enter this point after
                            // the first execution has finished and the
                            // shared locks have been deleted.
```

Reply:
Modified at [93f423b](https://github.com/iotaledger/iota/pull/4757/commits/93f423b897183cc74dc64b0552b8c58e860dc7cb).